pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -248,33 +248,22 @@ func (c *Cluster) removeExtraMgrs(daemonIDs []string) {
 	}
 }
 
-// UpdateActiveMgrLabel updates the mgr_role label value to either
-// active or standby depending on the status of the ceph mgr running
-// in the pod
-func (c *Cluster) UpdateActiveMgrLabel(daemonNameToUpdate string, prevActiveMgr string) (string, error) {
-
-	logger.Infof("Checking mgr_role label value of daemon %s (prev active mgr was %s)", daemonNameToUpdate, prevActiveMgr)
-	currActiveMgr, err := c.getActiveMgr()
-	if err != nil || currActiveMgr == "" {
-		logger.Infof("cannot update active mgr, no active mgr found. err=%v", err)
-		return "", err
-	} else if prevActiveMgr == currActiveMgr {
-		logger.Infof("active mgr is still the same (%s). No need to update mgr_role label on daemon %s.", currActiveMgr, daemonNameToUpdate)
-		return currActiveMgr, err
-	}
-
+// SetActiveMgrLabel sets the mgr_role label = active to currActiveMgr pod
+// and sets mgr_role label = standy to the rest of the mgr pods.
+func (c *Cluster) SetActiveMgrLabel(daemonNameToUpdate string, currActiveMgr string) error {
 	pods, err := c.context.Clientset.CoreV1().Pods(c.clusterInfo.Namespace).List(c.clusterInfo.Context, metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("mgr=%s", daemonNameToUpdate),
```

Comment:
I see that watcher lookups pods by `"mgr=<daemon name>"` label. And later on [line 270](https://github.com/rook/rook/pull/14482/files#diff-1881a7b5b13f1340101dddf0503eda118840efbda42a26c2cf6d640be74deb78R270) it compares `ceph_daemon_id` label instead of manager. 

In the cluster both of these labels have the same values:
```yaml
ceph_daemon_id: b
ceph_daemon_type: mgr
instance: b
mgr: b
mgr_role: active
```

Is there a reason to use both `mgr` and `ceph_daemon_id` labels? can they be different?

If these labels are equal, then this method can be simplified.

Reply:
For consistency, makes sense to update this method to use the same label instead of the two different labels. Those labels will always have the same value, so it would be the same behavior and no bug seen by the user in any case.